src/java/com/palantir/cassandra/utils/MutationVerificationHandler.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.cassandra.utils;
+
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.UnsafeArg;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.exceptions.InvalidMutationException;
+import org.apache.cassandra.utils.FBUtilities;
+import org.apache.cassandra.utils.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.List;
+
+public class MutationVerificationHandler implements OwnershipVerificationHandler
+{
+    private static final Logger logger = LoggerFactory.getLogger(OwnershipVerificationUtils.class);
+
+    public static final OwnershipVerificationHandler INSTANCE = new MutationVerificationHandler();
```

Comment:
why wouldn't this be of type `MutationVerificationHandler` and similarly for `ReadVerificationHandler`

Reply:
discussed offline, I'm just following the `List l = new ArrayList();` pattern where you always use the most generic interface where possible.